packages/block-editor/src/components/use-block-drop-zone/index.native.js

```diff
@@ -156,6 +159,15 @@ export default function useBlockDropZone( {
 			throttled.cancel();
 			targetBlockIndex.value = null;
 		},
+		onBlockDrop: ( event ) => {
+			if ( targetBlockIndex.value !== null ) {
+				onBlockDrop( {
+					...event,
+					targetRootClientId,
+					targetBlockIndex: targetBlockIndex.value,
+				} );
```

Comment:
The block drop event expects source and target parameters ([reference](https://github.com/WordPress/gutenberg/blob/f349616e8263e909c5b8283786b05b47142734fa/packages/block-editor/src/components/use-on-block-drop/index.native.js#L30-L35)), since the target is calculated by this hook, we already provide the target ones here.